gwnr/waveform/enigma_utils.py

```diff
@@ -442,11 +471,70 @@ def get_imr_enigma_modes(
         sample_rate=sample_rate,
         modes_to_use=modes_to_use,
         include_conjugate_modes=include_conjugate_modes,
-        return_orbital_params=False,  # These IMR functions don't have the functionality of returning the intermediate orbital parameters
+        return_orbital_params=return_orbital_params,
         verbose=verbose,
     )
 
     modes_numpy = retval[-1]
+    if return_orbital_params_user:
+        orb_var_dict = {
+            key: pt.TimeSeries(
+                retval[-2][key], delta_t=1.0 / sample_rate, epoch=retval[0][0]
+            )
+            for key in return_orbital_params_user
+        }
+
+    if f_window_mr_transition is None:
+        orb_freq = retval[-2]["phidot"] / ((mass1 + mass2) * lal.MTSUN_SI) / (2 * np.pi)
+        orb_phase = retval[-2]["phi"]
+
+        """Earlier implementation of the orbital cycle based hybridization that kept f_mr_transition at the hybridization frequency window's midpoint"""
+        # transition_idx = len(orb_freq) - np.argmax(
+        #     orb_freq[::-1] < f_mr_transition / 2.0
+        # )  # index at which orb_freq becomes just larger than transition orbital frequency, towards the end of waveform
+
+        #         if (orb_phase[transition_idx] + num_hyb_orbits * np.pi) > orb_phase[-1]:
+        #             raise Exception(
+        #                 f"""Requested number of orbits to hybridize over not available in the waveform after the transition frequency.
+        # Either decrease the number of orbits to hybridize over (currently {num_hyb_orbits}) or decrease the inspiral-to-merger transition frequency."""
+        #             )
+
+        #         window_start_idx = (
+        #             np.searchsorted(
+        #                 orb_phase, orb_phase[transition_idx] - num_hyb_orbits * np.pi
+        #             )
+        #             - 1
+        #         )  # phase is always a monotonically increasing function, hence using the more efficient np.searchsorted method
+        #         window_end_idx = np.searchsorted(
+        #             orb_phase, orb_phase[transition_idx] + num_hyb_orbits * np.pi
+        #         )
+        #         #         f_window_mr_transition = 2*(orb_freq[window_end_idx] - orb_freq[window_start_idx]) # 2 factor for returning the (2,2)-mode frequency
+        #         f_window_mr_transition = (
+        #             2
+        #             * np.max(
+        #                 [
+        #                     orb_freq[window_end_idx] - f_mr_transition / 2.0,
+        #                     f_mr_transition / 2.0 - orb_freq[window_start_idx],
+        #                 ]
+        #             )
+        #             * 2
+        #         )  # Extra 2-factor for returning the (2,2)-mode frequency
```

Comment:
Pls either delete this piece or put it behind a flag that a user can enable to specifically use hybridization windows centered around a frequency.

Reply:
Did that. In fact I've changed the default position of `f_mr_transition` at the end of the window only, regardless of whether the window is being fed by the user or is being calculated via the orbital cycle prescription. And I've added the flag `keep_f_mr_transition_at_center` if someone wants to keep it at the center only.